pkg/model/keycloak_deployment.go

```diff
@@ -349,3 +351,29 @@ func KeycloakVolumes() []v1.Volume {
 		},
 	}
 }
+
+// We allow the patch version of an image for keycloak to be modified outside of the operator on the cluster
+func getReconciledKeycloakImage(currentState *v13.StatefulSet) string {
+	currentImage := GetCurrentKeycloakImage(currentState)
+	currentImageRepo, currentImageMajor, currentImageMinor := getKeycloakImageRepoAndVersion(currentImage)
+	keycloakImageRepo, keycloakImageMajor, keycloakImageMinor := getKeycloakImageRepoAndVersion(KeycloakImage)
+
+	// Check the repos, major and minor versions match. If so, return and reconcile with the current cluster image
+	// E.g. quay.io/keycloak/keycloak:7.0.1
+	if currentImageRepo == keycloakImageRepo && currentImageMajor == keycloakImageMajor && currentImageMinor == keycloakImageMinor {
+		return currentImage
+	}
+
+	return KeycloakImage
+}
+
+// Split a full keycloak image string (e.g. quay.io/keycloak/keycloak:7.0.1) into it's repo and individual versions
+func getKeycloakImageRepoAndVersion(image string) (string, string, string) {
+	imageStrings := strings.Split(image, ":")
+	imageRepo, imageTag := imageStrings[0], imageStrings[1]
+
+	imageTagStrings := strings.Split(imageTag, ".")
+	imageMajor, imageMinor := imageTagStrings[0], imageTagStrings[1]
```

Comment:
Also, I think we might want to add some checks (or fallbacks) if we use tags like `latest` or `master`. In that case `imageTagStrings[1]` won't exist.

Reply:
Thank you, fixed here - https://github.com/keycloak/keycloak-operator/pull/70